src/garage/tf/algos/reps.py

```diff
@@ -68,13 +71,23 @@ def __init__(self,
                  l2_reg_dual=0.,
                  l2_reg_loss=0.,
                  optimizer=LbfgsOptimizer,
-                 optimizer_args=dict(max_opt_itr=50),
+                 optimizer_args=None,
                  dual_optimizer=scipy.optimize.fmin_l_bfgs_b,
-                 dual_optimizer_args=dict(maxiter=50),
+                 dual_optimizer_args=None,
                  name='REPS'):
+        optimizer_args = optimizer_args or dict(max_opt_itr=50)
+        dual_optimizer_args = dual_optimizer_args or dict(maxiter=50)
+
         self.name = name
         self._name_scope = tf.name_scope(self.name)
 
+        self.feat_diff = None
```

Comment:
can you make all of these attrs _private while you are visiting?

Reply:
Fixed. Also did so for NPO.py